src/pty.rs

```diff
@@ -0,0 +1,128 @@
+//! Create master and slave virtual pseudo-terminals (PTTYs)
+
+use std::ffi::CStr;
+use std::os::unix::io::RawFd;
+
+use libc;
+
+use {Error, fcntl, Result};
+
+/// Grant access to a slave pseudoterminal (see
+/// [grantpt(3)](http://man7.org/linux/man-pages/man3/grantpt.3.html))
+///
+/// `grantpt()` changezs the mode and owner of the slave pseudoterminal device corresponding to the
+/// master pseudoterminal referred to by `fd`. This is a necessary step towards opening the slave.
+#[inline]
+pub fn grantpt(fd: RawFd) -> Result<()> {
```

Comment:
I don't think it should implement `From<RawFd>`, because that would make it too easy to accidentally convert an inappropriate value.  Conversion should always be explicit.  In fact, are there any other Rust functions, in `nix` or in `std`, that deal with pseudoterminals?